view/src/main/java/cmm/apps/esmorga/view/eventdetails/EventDetailsViewModel.kt

```diff
@@ -80,17 +83,36 @@ class EventDetailsViewModel(
             _uiState.value = _uiState.value.copy(primaryButtonLoading = true)
             val result = joinEventUseCase(eventId)
             result.onSuccess {
+                userJoined = true
                 _uiState.value = _uiState.value.copy(primaryButtonLoading = false, primaryButtonTitle = getPrimaryButtonTitle(true, true))
-                _effect.tryEmit(EventDetailsEffect.ShowJoinEventSuccessSnackbar)
+                _effect.tryEmit(EventDetailsEffect.ShowJoinEventSuccess)
             }.onFailure { error ->
-                _uiState.value = _uiState.value.copy(primaryButtonLoading = false)
-                if (error.code == ErrorCodes.NO_CONNECTION) {
-                    _effect.tryEmit(EventDetailsEffect.ShowNoNetworkError())
-                } else {
-                    _effect.tryEmit(EventDetailsEffect.ShowFullScreenError())
-                }
+                showErrorScreen(error)
             }
         }
     }
 
+    private fun leaveEvent() {
+        viewModelScope.launch {
+            _uiState.value = _uiState.value.copy(primaryButtonLoading = true)
+            val result = leaveEventUseCase(eventId)
+            result.onSuccess {
+                userJoined = false
+                _uiState.value = _uiState.value.copy(primaryButtonLoading = false, primaryButtonTitle = getPrimaryButtonTitle(true, false))
```

Comment:
the call to `getPrimaryButtonTitle` would be clearer if you include the name of the parameters in it.